phpstan.neon

```diff
@@ -14,6 +14,7 @@ parameters:
         - '/^Return typehint of method PhpCsFixer\\Tests\\Test\\.+::createIsIdenticalStringConstraint\(\) has invalid type PHPUnit_Framework_Constraint_IsIdentical\.$/'
         - '/^Class (Symfony\\Contracts\\EventDispatcher\\Event|Symfony\\Component\\EventDispatcher\\Event) not found.$/'
         - '/^Constant T_NAME_(RELATIVE|FULLY_QUALIFIED|QUALIFIED) not found\.$/'
+        - '/Static call to instance method PhpCsFixer\\Tests\\TestCase::assert.*/'
```

Comment:
Why do we need this?

Reply:
Because PHPStan will fail the build without it. And _why_ it thinks we may call any `assert*` instance methods (where we know they're all `static`) is beyond me.

Reply:
Not a bug. It'll be caused by the `__call` method you have?

Reply:
If you take a look at the header of `DeprecatedTestMethods` you'll notice that all `assert*` methods declared there are `static`. Here's [a similar issue](https://github.com/phpstan/phpstan/issues/1557), and a proposed workaround there is the same used here.

Reply:
We need `__call` for `expectException*` methods. They're indeed instance methods. PHPStan has nothing against them.

Reply:
Just for the record, removing `__call` does nothing to the issue. These are still there:

```
 ------ ----------------------------------------------------------------------------- 
  Line   tests/RuleSet/Sets/AbstractSetTest.php                                       
 ------ ----------------------------------------------------------------------------- 
  66     Static call to instance method PhpCsFixer\Tests\TestCase::assertIsString().  
 ------ ----------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------------------------------- 
  Line   tests/Test/AbstractFixerTestCase.php                                                         
 ------ --------------------------------------------------------------------------------------------- 
  484    Static call to instance method PhpCsFixer\Tests\TestCase::assertIsString().                  
  485    Static call to instance method PhpCsFixer\Tests\TestCase::assertMatchesRegularExpression().  
  486    Static call to instance method PhpCsFixer\Tests\TestCase::assertStringNotContainsString().   
 ------ --------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------- 
  Line   tests/Tokenizer/TokensTest.php                                              
 ------ ---------------------------------------------------------------------------- 
  1411   Static call to instance method PhpCsFixer\Tests\TestCase::assertIsArray().  
  1418   Static call to instance method PhpCsFixer\Tests\TestCase::assertIsArray().  
 ------ ---------------------------------------------------------------------------- 
```
You can [check with `TokensTest.php`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3f1b03d123898e1a5a0b9fba3c6f2cb7d4fc39c5/tests/Tokenizer/TokensTest.php#L1411) just for an example, to see that PHPStan is telling a pure BS here.

Reply:
Hmmm. Can we come up with a minimal example, so we can report the issue to them?

Reply:
Vanilla PHPStan has nothing like that: https://phpstan.org/r/16b02a02-a9b5-4b65-9aa2-30a0e4ffd586

Reply:
Commenting out `phpstan/phpstan-phpunit/extension.neon` alleviates the issue. If you wish to report the issue to `phpstan/phpstan-phpunit`, please do. I'm not going to have the time to follow them up.